sw/device/tests/keymgr_sideload_otbn_test.c

```diff
@@ -100,11 +133,49 @@ static void test_otbn_with_sideloaded_key(dif_keymgr_t *keymgr,
 
   // Load the X25519 application.
   CHECK(otbn_load_app(otbn_ctx, kOtbnAppX25519) == kOtbnOk);
-
   // Run the OTBN app and retrieve the result.
   uint32_t result[8];
   CHECK(run_x25519_app(otbn_ctx, result) == true);
 
+  // Clear the sideload key and check that OTBN errors with the correct error
+  // code (`KEY_INVALID` bit 5 = 1).
+  CHECK_DIF_OK(
+      dif_keymgr_sideload_clear_set_enabled(keymgr, kDifToggleEnabled));
+  LOG_INFO("Clearing the Keymgr generated sideload keys.");
+  uint32_t at_clear_salt_result[8];
+  CHECK(run_x25519_app_clearkey(otbn_ctx, at_clear_salt_result) == false);
+  CHECK_ARRAYS_NE(result, at_clear_salt_result, ARRAYSIZE(result));
+
+  // Disable sideload key clearing
+  CHECK_DIF_OK(
+      dif_keymgr_sideload_clear_set_enabled(keymgr, kDifToggleDisabled));
+  LOG_INFO("Disable clearing the Keymgr generated sideload keys.");
+
+  // Clear the ERR bits REG and reset the OTBN interrupts (should not even be
+  // necessary for this recoverable software error)
+  mmio_region_write32(otbn_ctx->dif.base_addr, OTBN_ERR_BITS_REG_OFFSET, 0x0);
+  CHECK_DIF_OK(dif_otbn_reset(&otbn_ctx->dif));
+
+  // Regenerate the sideload key
+  keymgr_testutils_generate_versioned_key(keymgr, sideload_params);
+  LOG_INFO("Keymgr generated HW output for OTBN.");
+
+  // Confirm keymgr is in idle status and without error
+  dif_keymgr_status_codes_t keymgr_status;
+  CHECK_DIF_OK(dif_keymgr_get_status_codes(keymgr, &keymgr_status));
+  LOG_INFO("Keymgr Status: 0x%08x", keymgr_status);
```

Comment:
sounds like `keymgr_testutils_generate_versioned_key` already invokes `dif_keymgr_get_status_codes` to check it doesn't have any error. perhaps we can delete this portion.